### PR TITLE
Migrate BashOperator guide

### PR DESCRIPTION
**Pull request by jwitz**

### CODE REVIEW

**Comment by kwcanuck** [comment]
@jwitz it appears that you've selected _main_ and not _guide-migration-base_ as the merge branch. Is this correct?

**Comment by jwitz** [comment]
@kwcanuck I updated as you were writing your comment! If you notice this in the future, feel free to update the base branch yourself - otherwise I'll probably remember within 20 minutes of making the PR haha

File: sidebarsGuides.js [hunk 1]
```diff
@@ -32,13 +32,15 @@ module.exports = {
...
         'custom-xcom-backends',
+        'bashoperator',
       ],
     },
     {
       type: 'category',
       label: 'Tutorials',
       items: [
         'overview',
+        'executing-notebooks'
```
Comment (jwitz) [inline sidebarsGuides.js:43]:
```suggestion
```

File: sidebarsGuides.js [hunk 2]
```diff
@@ -57,6 +59,7 @@ module.exports = {
         'airflow-talend-integration',
         'airflow-sql-data-quality-tutorial',
         'astro-python-sdk-etl',
+        'aws-lambda'
```
Comment (jwitz) [inline sidebarsGuides.js:62]:
```suggestion
```

File: guides/bashoperator.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+sidebar_label: Using the BashOperator
+---
+
+The [BashOperator](https://registry.astronomer.io/providers/apache-airflow/modules/bashoperator) is one of the most commonly used operators in Airflow. It executes bash commands or a bash script from within your Airflow DAG.
+
+In this guide you'll learn:
+
+- When to use the BashOperator.
+- How to use the BashOperator.
+- Examples of how to use the BashOperator including executing bash commands and bash scripts.
```
Comment (kwcanuck) [inline guides/bashoperator.md:14]:
```suggestion
- How to use the BashOperator including executing bash commands and bash scripts.
```

File: guides/bashoperator.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+- Running scripts in a programming language other than Python.
+- Running commands kicking off tools that do not have specific operator support yet. For example [Soda Core](https://www.astronomer.io/guides/soda-data-quality/).
+
+## Example: Execute two bash commands using one BashOperator
+
+The BashOperator can execute any number of bash commands separated by `&&`.
+
+In this example, you run two bash commands in a single task:
+
+- `echo Hello $MY_NAME!` prints the environment variable `MY_NAME` to the console.
```
Comment (kwcanuck) [inline guides/bashoperator.md:66]: We're using _terminal_, _terminal window_, and _console_ interchangeably. We should pick one option and use it consistently.

File: guides/bashoperator.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+
+Make sure that your bash script (`my_bash_script.sh` in this example) is available to your Airflow environment. If you use the Astro CLI, you can make this file accessible to Airflow by placing it in the `/include` directory of your Astro project.
+
+It is important to make the bash script executable by running the following command before making the script available to your Airflow environment:
+
+```bash
+chmod +x my_bash_script.sh
+```
+
+If you use the Astro CLI, you can either run this command before running `astro dev start`, or you can add the command to your project's Dockerfile with the following `RUN` command:
```
Comment (kwcanuck) [inline guides/bashoperator.md:127]:
```suggestion
If you use the Astro CLI, you can run this command before running `astro dev start`, or you can add the command to your project's Dockerfile with the following `RUN` command:
```

File: guides/bashoperator.md [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+        bash_command="$AIRFLOW_HOME/include/my_bash_script.sh "
+        # since the env argument is not specified, this instance of the
+        # BashOperator has access to the environment variables of the Airflow
+        # instance like AIRFLOW_HOME
+    )
+```
+
+## Example: Run a script in another programming language
+
+Using the BashOperator is the easiest way to run a script in a non-Python programming language in Airflow. You can run a script in any language that can be run with a bash command.
```
Comment (kwcanuck) [inline guides/bashoperator.md:163]:
```suggestion
Using the BashOperator is a straightforward way to run a script in a non-Python programming language in Airflow. You can run a script in any language that can be run with a bash command.
```

File: guides/bashoperator.md [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+
+The following setup is required:
+
+- Install the JavaScript and R language packages at the OS level.
+- Write a JavaScript file.
+- Write a R script file.
+- Make the scripts available to the Airflow environment.
+- Execute the files from within a DAG using the BashOperator.
+
+If you use the Astro CLI, the programming language packages can be installed at the OS level by specifying them to the `packages.txt` file of your Astro project.
```
Comment (kwcanuck) [inline guides/bashoperator.md:175]:
```suggestion
If you use the Astro CLI, the programming language packages can be installed at the OS level by adding them to the `packages.txt` file of your Astro project.
```

File: guides/bashoperator.md [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+// in case of an error print the error statement to the console
+req.on('error', error => {
+  console.error(error);
+});
+
+req.end();
+```
+
+
+The second task runs a script written in R that uses a regex to filter the longitude and latitude information from the API response and print it in a sentence.
```
Comment (kwcanuck) [inline guides/bashoperator.md:215]:
```suggestion
The second task runs a script written in R that uses a regex to filter and print the longitude and latitude information from the API response.
```

File: guides/bashoperator.md [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+        },
+        # set append_env to True to be able to use env variables
+        # like AIRFLOW_HOME from the Airflow environment
+        append_env=True
+    )
+
+    get_ISS_coordinates >> print_ISS_coordinates
+```
+
+The logs from the second task will show a statement with the current ISS location:
```
Comment (kwcanuck) [inline guides/bashoperator.md:275]:
```suggestion
The logs from the second task include a statement with the current ISS location:
```

**Review by kwcanuck — APPROVED** [review]

File: guides/bashoperator.md [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+- Running commands kicking off tools that do not have specific operator support yet. For example [Soda Core](https://www.astronomer.io/guides/soda-data-quality/).
+
+## Example: Execute two bash commands using one BashOperator
+
+The BashOperator can execute any number of bash commands separated by `&&`.
+
+In this example, you run two bash commands in a single task:
+
+- `echo Hello $MY_NAME!` prints the environment variable `MY_NAME` to the console.
+- `echo $A_LARGE_NUMBER | rev  2>&1 | tee $AIRFLOW_HOME/include/my_secret_number.txt` takes the environment variable `A_LARGE_NUMBER`, pipes it to the `rev` command which reverses any input, and saves the result in a file called `my_secret_number.txt` located in the `/include` directory. The reversed number will also be printed to the terminal.
```
Comment (jwitz) [inline guides/bashoperator.md:67]:
```suggestion
- `echo $A_LARGE_NUMBER | rev  2>&1 | tee $AIRFLOW_HOME/include/my_secret_number.txt` takes the environment variable `A_LARGE_NUMBER`, pipes it to the `rev` command which reverses any input, and saves the result in a file called `my_secret_number.txt` located in the `/include` directory. The reversed number will also be printed to the console.
```

**Pull request merged**